Controls/BootstrapDatepicker/src/DotVVM.Contrib/BootstrapDatepicker.cs

```
@@ -0,0 +1,104 @@
+        {
+            context.ResourceManager.AddCurrentCultureGlobalizationResource();
+
+            IValueBinding dateBinding = null;
+            foreach (var item in properties)
+                if (item.Key == DateProperty)
+                    dateBinding = item.Value as IValueBinding;
+
+            if (dateBinding == null)
+            {
+                var expression = dateBinding.GetKnockoutBindingExpression(this);
+                expression = "dotvvm.globalize.formatString(" + JsonConvert.ToString("d.M.yyyy") + ", " + expression + ")";
+                writer.AddKnockoutDataBind("dotvvm-contrib-BootstrapDatepicker", expression);
+            }
+            else
+            {
+                writer.AddKnockoutDataBind("dotvvm-contrib-BootstrapDatepicker", this, DateProperty, renderEvenInServerRenderingMode: true);
+            }
+
+            if (!string.IsNullOrWhiteSpace(Language))
+                writer.AddAttribute("data-date-language", Language);
+
+            base.AddAttributesToRender(writer, context);
+        }
+
+        protected override void OnPreRender(IDotvvmRequestContext context)
+        {
+            context.ResourceManager.AddRequiredResource("dotvvm.contrib.BootstrapDatepicker");
+
+            var language = Language;
+
+            if (DotvvmConfigurationExtensions.CurrentLocales != null)
+            {
+                if (string.IsNullOrWhiteSpace(language))
+                {
+                    language = CultureInfo.CurrentUICulture.Name;
+                    if (!DotvvmConfigurationExtensions.CurrentLocales.Any(p => p.Equals(language, StringComparison.OrdinalIgnoreCase)))
+                    {
+                        language = language.Substring(0, 2);
+                        if (!DotvvmConfigurationExtensions.CurrentLocales.Any(p => p.Equals(language, StringComparison.OrdinalIgnoreCase)))
+                            language = null;
```

Well, do you think that just assuming default language is a good idea when there it's not found?

---

Do you think, it would be better to throw the exception "language not found"?

---

Yes definitely. In case someone assigns some wrong value (it may be a typo or the he forgot to register it in the configuration), it should rather throw an exception that something is wrong than silently eating the error.